plugins/modules/vm_resource_info.py

```diff
@@ -0,0 +1,315 @@
+            'consumed_overhead_mb': vm.summary.quickStats.consumedOverheadMemory,
+            'guest_usage_mb': vm.summary.quickStats.guestMemoryUsage,
+            'host_usage_mb': vm.summary.quickStats.hostMemoryUsage,
+            'swapped_mb': vm.summary.quickStats.swappedMemory,
+            'distributed_entitlement_mb': vm.summary.quickStats.distributedMemoryEntitlement,
+            'static_entitlement_mb': vm.summary.quickStats.staticMemoryEntitlement
+        }
+
+    def _format_esxi_output(self, vm):
+        try:
+            return {
+                'name': vm.runtime.host.name,
+                'moid': vm.runtime.host._GetMoId()
+            }
+        except AttributeError:
+            return {}
+
+    def _format_rp_output(self, vm):
+        try:
+            return {
+                'name': vm.resourcePool.name,
+                'moid': vm.resourcePool._GetMoId()
+            }
+        except AttributeError:
+            return {}
+
+    def gather_info_for_vms(self):
+        all_vm_info = []
+        for vm in self.get_vms():
+            info = {
+                'name': vm.name,
+                'moid': vm._GetMoId(),
+                'esxi_host': self._format_esxi_output(vm),
+                'resource_pool': self._format_rp_output(vm),
+                'cpu': self._format_cpu_output(vm),
+                'memory': self._format_memory_output(vm),
+                'stats': {
+                    'cpu': self._format_cpu_stats_output(vm),
+                    'memory': self._format_memory_stats_output(vm)
+                }
+            }
+            all_vm_info += [info]
+
+        return all_vm_info
+
+    def get_vms(self):
+        """
+        Uses the UUID, MOID, or name provided to find the source VM for the template. Returns an error if using the name,
+        multiple matches are found, and the user did not provide a name_match strategy.
+        """
+        if self.params.get('name') or self.params.get('uuid') or self.params.get('moid'):
+            vm = self.get_vm_using_params(fail_on_missing=False)
+        else:
+            vm = self.get_all_vms()
+
+        return vm if vm else []
+
+
+def main():
+    argument_spec = base_argument_spec()
+    argument_spec.update(
+        dict(
+            name=dict(type='str'),
+            name_match=dict(type='str', choices=['first', 'last'], default=None),
+            uuid=dict(type='str'),
+            use_instance_uuid=dict(type='bool', default=True),
+            moid=dict(type='str'),
+            gather_cpu_config=dict(type='bool', default=True),
+            gather_memory_config=dict(type='bool', default=True),
+            gather_cpu_stats=dict(type='bool', default=True),
+            gather_memory_stats=dict(type='bool', default=True),
+        )
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[['name', 'uuid', 'moid']]
```

Comment:
None of them are required? Will it return information about all VMs if a specific VM is not specified?